frontend/model/state.js

```diff
@@ -329,6 +335,235 @@ const getters = {
       return profile.displayName || username
     }
   },
+  // used with graphs like those in the dashboard and in the income details modal
+  groupIncomeDistribution (state, getters) {
+    return groupIncomeDistribution({ state, getters, monthstamp: currentMonthstamp() })
+  },
+  // adjusted version of groupIncomeDistribution, used by the payments system
+  groupIncomeAdjustedDistribution (state, getters) {
+    return getters.groupIncomeAdjustedDistributionForMonth(currentMonthstamp())
+  },
+  groupIncomeAdjustedDistributionForMonth (state, getters) {
+    return monthstamp => groupIncomeDistribution({ state, getters, monthstamp, adjusted: true })
+  },
+  ourPayments (state, getters) {
+    const currency = currencies[getters.groupSettings.mincomeCurrency]
+    const ourUsername = getters.ourUsername
+    const cMonthstamp = currentMonthstamp()
+    const pDate = dateFromMonthstamp(cMonthstamp)
+    const dueIn = lastDayOfMonth(pDate)
+    const monthlyPayments = getters.groupMonthlyPayments
+    const allPayments = getters.currentGroupState.payments
+
+    const sent = (() => {
+      const payments = []
+      // QUESTION: Is this only from this month? What about the previous payments from months ago?
+      //           Same question for received.
+      for (const monthstamp of Object.keys(monthlyPayments).sort()) {
+        const { paymentsFrom } = monthlyPayments[monthstamp]
+
+        if (!paymentsFrom) { continue }
+
+        for (const toUser in paymentsFrom[getters.ourUsername]) {
+          for (const paymentHash of paymentsFrom[getters.ourUsername][toUser]) {
+            const { data, meta } = allPayments[paymentHash]
+
+            payments.push({ hash: paymentHash, data, meta })
+          }
+        }
+      }
+      return payments
+    })()
+    const received = (() => {
+      const thisMonthPayments = monthlyPayments[cMonthstamp]
+      const paymentsFrom = thisMonthPayments && thisMonthPayments.paymentsFrom
+      const payments = []
+
+      if (!paymentsFrom) { return [] }
```

Comment:
Same here. There would be a single `return` statement in this function if this was restored to how it was before (with the `if` wrapping the code below).

Same goes for lines 403 and 408 below. It would make it easier for me to read and follow the logic of the code, as well as compare it to the previous code to see if anything significant changed. Would appreciate if that style/logic can be revered.

Reply:
Hum... can we leave as it is? This avoids having to many "nested conditions"

Reply:
I would prefer it be restored to how it was before, because it represents a functional style that's easier to follow and modify. For example, the previous version of this code makes it easier to add an `else { .. }` condition in a natural fashion. Having a single `return` for the function also makes it easier to reason about how and when the code will exit from its task. That is why I wrote the code that way, so I'd really prefer it to be how it was written originally if it's not a big deal for you.

Reply:
I understand. reverted

Reply:
🙇